### PR TITLE
Support S3 forcePathStyle for systems like minio

### PR DESCRIPTION
**Pull request by mbessette94**
- Added .devcontainer for vscode development in docker
- Added support for `forcePathStyle` on S3 Driver
- Tested with minio uploading Avatars and Files onto collection records

Let me know if you'd prefer the toggle moved
Screenshot from 2022-07-17 21-46-03

### CODE REVIEW

File: core/settings.go
Comment (ganigeorgiev) [inline core/settings.go:264]: This line could be removed. The `ForcePathStyle` should be optional, not required.

File: ui/src/components/settings/PageStorage.svelte
Comment (ganigeorgiev) [inline ui/src/components/settings/PageStorage.svelte:138]: This checkbox should be optional and not required.
Let's also move it below under all the other S3 fields and make it a regular checkbox (aka. remove the `form-field-toggle` class).
Let's also remove the question mark from the label and add a info icon with tooltip explaining what this option do.
Or in other words, place the below code, before `<!-- margin helper -->`
```html
// add the tooltip action in the imports section
import tooltip from "@/actions/tooltip";

....

<div class="col-lg-12">
    <Field class="form-field" name="s3.forcePathStyle" let:uniqueId>
        <input type="checkbox" id={uniqueId} bind:checked={s3.forcePathStyle} />
        <label for={uniqueId}>
            <span class="txt">Force path-style addressing</span>
            <i
                class="ri-information-line link-hint"
                use:tooltip={{
                    text: 'Enable to force the request to use path-style addressing, eg. "https://s3.amazonaws.com/BUCKET/KEY" instead of the default "https://BUCKET.s3.amazonaws.com/KEY".',
                    position: "top",
                }}
            />
        </label>
    </Field>
</div>
```

File: .devcontainer/Dockerfile
Comment (ganigeorgiev) [inline .devcontainer/Dockerfile:18]: Let's not add this since it is too specific to the user environment and for vscode only (there also too many optional things that I haven't tested).

File: .devcontainer/devcontainer.json
Comment (ganigeorgiev) [inline .devcontainer/devcontainer.json:58]: Same as above.

We may add it in the future, but let's keep it out of this PR.

**Review by ganigeorgiev — CHANGES_REQUESTED** [review]
Hi, thanks for the PR!

I left some notes, you could review them when you have time.

I'm usually against this option, because it is [deprecated by amazon](https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/) and probably they will remove it from the library too in future versions, but OK. Let's keep it for better compatibility with other S3 providers.

**Comment by ganigeorgiev** [comment]
In addition, also make sure to run/update the Go tests - see https://github.com/pocketbase/pocketbase/blob/master/CONTRIBUTING.md#making-changes-in-the-go-code

File: ui/src/components/settings/PageStorage.svelte
Comment (Lex-2008) [inline ui/src/components/settings/PageStorage.svelte:136]: Maybe also remove question mark here?

Note that I'm not a maintainer here, so up to @ganigeorgiev to decide :-)

But I think that usually options don't have question marks - see, for example, "Use S3 storage" option on your screenshot.

**Review by Lex-2008 — COMMENTED** [review]
nice! Without it I had to use `MINIO_DOMAIN` env variable and add bucket name to DNS (or /etc/hosts), see #90. And with this option it will just work!

File: ui/src/components/settings/PageStorage.svelte
Comment (Matt-Bessette) [reply ui/src/components/settings/PageStorage.svelte]: Updated per @ganigeorgiev's phrasing of it :D I'm not much of a ux person, I'm used to being told where to put stuff!

**Comment by ganigeorgiev** [comment]
Thank you!

I've merged it locally since the tests had to be updated. It will be included in the next minor bugfix/feature release.

**Pull request closed without merging**
